### PR TITLE
Alex post page prototype: Flask + Jinja2 + MySQL Connector

### PR DESCRIPTION
**Pull request by abmokin**
For the prototype post, I've connected together Flask, Jinja2 and MySQL Connector.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from flask import Flask, render_template
+import mysql.connector
+from datetime import date, datetime, timedelta
+import os
+
+# Flask: Create Flask object
+app = Flask(__name__)
+
+# MySQL Connector: Create cursor
+cnx = mysql.connector.connect(host='localhost', port=3306, user='root', passwd='my#Blog!db456', db='blog')
```
Comment (mmotorny) [inline main.py]: Password needs to be passed on command line.

Comment (abmokin) [reply main.py]: No way!

**Comment by abmokin** [comment]
String with a great wisdom comment

File: main.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import os
+import sys
+from datetime import date, datetime, timedelta
+from jinja2 import Environment, FileSystemLoader
+from flask import Flask, render_template
+import mysql.connector
+
+
+if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (abmokin) [reply main.py]: OK, I got it

File: main.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    # Flask: Create Flask object
+    app = Flask(__name__)
+
+    # MySQL Connector: Create cursor
+    cnx = mysql.connector.connect(host=sys.argv[1], port=int(sys.argv[2]), user=sys.argv[3],
```
[the remark this answers is not in the document]
Comment (abmokin) [reply main.py]: I see

File: main.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        else:
+            column_name_str += column_name_raw[0]
+
+    # MySQL: Read data from the 'titles' table
+
+    # query_read_titles = ("SELECT "
+    #                      "title_main_blog, title_blog_name, title_main_description, title_comments, "
+    #                      "title_form_comment_text, title_form_comment_name, title_form_comment_email, "
+    #                      "title_form_comment_submit, title_footer} "
+    #                      "FROM titles")
```
[the remark this answers is not in the document]
Comment (abmokin) [reply main.py]: Got it. I did it to ask which option is better.

File: main.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    # Jinja2: Create dictionary with the values from the 'posts' table
+    post_dict = {}
+    for post_date, post_title, post_text in cursor:
+        post_dict['post_title'] = post_title
+        post_dict['post_date'] = post_date
+        post_dict['post_text'] = post_text
+
+    # MySQL: Read column names of the 'titles' table
+    query_read_column_names = ("SELECT `COLUMN_NAME` "
```
[the remark this answers is not in the document]
Comment (abmokin) [reply main.py]: OK, thanks!

File: main.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    #                      "title_form_comment_submit, title_footer} "
+    #                      "FROM titles")
+    query_read_titles = ("SELECT {} FROM titles".format(column_name_str))
+    cursor.execute(query_read_titles)
+
+    # Jinja2: Add to dictionary values from the 'titles' table
+    counter = 0
+    for titles in cursor:
+        for title in titles:
+            post_dict[column_name_list[counter]] = title
```
[the remark this answers is not in the document]
Comment (abmokin) [reply main.py]: OK, I'll try.

File: main.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    # Jinja2: Capture directory of the template
+    template_dir = os.path.dirname(os.path.abspath(__file__)) + '/templates/'
+
+    # Jinja2: Create Environment object
+    j2_env = Environment(loader = FileSystemLoader(template_dir), trim_blocks = True)
+
+    # Flask: Create main endpoint
+    @app.route('/')
+    def index():
+        return j2_env.get_template('main.html').render(post_dict)
```
[the remark this answers is not in the document]
Comment (abmokin) [reply main.py]: I am not sure that I understand you right, but I'll try to fix it.

**Pull request closed without merging**